### DISCUSSION TITLE
Add Feature to Chat with ChatGPT / LLM

### DISCUSSION THREAD

**Discussion by grabani** [pawelsalawa/sqlitestudio#4919] [Ideas]
I constantly find myself having to leave SQLite Studio (Studio) and going over to a browser and chatting with ChatGPT to determine SQL queries which I may use (within Studio). In the first instance, it would be super useful to add a menu option which allow you to chat directly with ChatGPT (via API) to obtain help with determining a SQL query. Later on this capability may be enhanced so that Studio would provide shortcuts to copy things like table schemas into the chat window (but that would be for a later phase).

In my naiveté I do not think this wouldn't be a complex feature to add. The ChatGPT API doc on Chat requests can be found here: [https://platform.openai.com/docs/api-reference/chat/create](url)

**Comment by pawelsalawa** [comment]
Are you using the paid version of ChatGPT?

**Reply by grabani** [reply]
I have both. I use the browser based for general GPT-3.5 model ([chat.openai.com](url)) and also I have a separate GUI (opensource browser based) for chatting using my API key (to access the GPT-4 model).

**Reply by pawelsalawa** [reply]
Okay. I asked, because API is available only in paid version of ChatGPT.
I've created #4920

**Reply by grabani** [reply]
so are we doing this???!!! 😊

**Reply by pawelsalawa** [reply]
Some day...

**Reply by grabani** [reply]
if you give me some pointers I can see if I can fork and try it myself for this repo. For example:

1) How do I add a menu and sub menu item where are the functions (code) for this if I want to add new ones.
2) where are sub menu functions located (I am assuming separate function for each sub menu)
3) Where is the function / code that opens a tab at the bottom of the UI; similar to when you choose "Tool"s --> "Open SQL Editor"?


I guess a more fundamental question is, would this functionality be added as a plug-in or as part of the core code?

**Reply by pawelsalawa** [reply]
Ad. 1 & 2) I'm guessing you mean the top menubar, in which case the code for it is at https://github.com/pawelsalawa/sqlitestudio/blob/master/SQLiteStudio3/guiSQLiteStudio/mainwindow.cpp#L323

Ad. 3)  You can see here https://github.com/pawelsalawa/sqlitestudio/blob/master/SQLiteStudio3/guiSQLiteStudio/mainwindow.cpp#L928 how MDI windows are open - there is a shortcut function that does that for you, as long as your MDI window extends the `MdiChild` class. You can see plenty examples of such MDI windows in https://github.com/pawelsalawa/sqlitestudio/tree/master/SQLiteStudio3/guiSQLiteStudio/windows

**Reply by pawelsalawa** [reply]
> I guess a more fundamental question is, would this functionality be added as a plug-in or as part of the core code?
Plugin would be more preferrable obviously, but core functionality is fine too - it's not that hard to modify it into a plugin later on.

**Reply by grabani** [reply]
Wish me luck. I may need your help. Hope that is ok.
